src/Features/Core/Portable/EmbeddedLanguages/Json/JsonLexer.cs

```diff
@@ -0,0 +1,403 @@
+        }
+
+        private ImmutableArray<JsonTrivia> ScanTrivia(bool leading)
+        {
+            var result = ArrayBuilder<JsonTrivia>.GetInstance();
+
+            while (Position < Text.Length)
+            {
+                var comment = ScanComment();
+                if (comment != null)
+                {
+                    result.Add(comment.Value);
+                    continue;
+                }
+
+                var endOfLine = ScanEndOfLine();
+                if (endOfLine != null)
+                {
+                    result.Add(endOfLine.Value);
+
+                    if (leading)
+                    {
+                        continue;
+                    }
+                    else
+                    {
+                        break;
+                    }
+                }
+
+                var whitespace = ScanWhitespace();
+                if (whitespace != null)
+                {
+                    result.Add(whitespace.Value);
+                    continue;
+                }
+
+                break;
+            }
+
+            return result.ToImmutableAndFree();
+        }
+
+        private JsonTrivia? ScanEndOfLine()
+        {
+            var start = Position;
+            if (IsAt("\r\n"))
+            {
+                Position += 2;
+                return CreateTrivia(JsonKind.EndOfLineTrivia, GetCharsToCurrentPosition(start));
+            }
+            else if (IsAt("\r") || IsAt("\n"))
+            {
+                Position++;
+                return CreateTrivia(JsonKind.EndOfLineTrivia, GetCharsToCurrentPosition(start));
+            }
+
+            return null;
+        }
+
+        public JsonTrivia? ScanComment()
+        {
+            if (IsAt("//"))
+            {
+                return ScanSingleLineComment();
+            }
+            else if (IsAt("/*"))
+            {
+                return ScanMultiLineComment();
+            }
+            else if (IsAt("/"))
+            {
+                var start = Position;
+                Position++;
+
+                var chars = GetCharsToCurrentPosition(start);
+                return CreateTrivia(JsonKind.SingleLineCommentTrivia, chars,
+                    ImmutableArray.Create(new EmbeddedDiagnostic(
+                        FeaturesResources.Error_parsing_comment,
+                        GetSpan(chars))));
+            }
+
+            return null;
+        }
+
+        private JsonTrivia ScanSingleLineComment()
+        {
+            var start = Position;
+            Position += 2;
+
+            while (Position < Text.Length &&
+                   this.CurrentChar is var ch &&
+                   ch != '\r' && ch != '\n')
+            {
+                Position++;
+            }
+
+            var chars = GetCharsToCurrentPosition(start);
+            if (Position == start + 2)
+            {
+                // Note: json.net reports an error if the file ends with "//", so we just
```

Comment:
but... why does it do that? :(

Reply:
who knows :)  it is fickle and a weird impl.  comments aren't legal in json to begin with.... whatchagoingtodo ? :)

Reply:
I assume the parser supports `//` as a single line on it's own as long as it's not the last (that's my understanding from this code). That's just weird... idk if we even want to keep that behavior? probably just a nit in the end though